Server.py

```diff
@@ -0,0 +1,226 @@
+
+    def terminate(self):
+        self._run = False
+
+    def add_server(
+        self,
+        our_address=("localhost", 80)
+    ):
+        """ Creat server side socket and add it to the database. \n
+            our_address: tupple of the address that the server will do bind to.
+                         format (address, port), default (localhost, 80)
+        """
+        s = util.creat_nonblocking_socket()
+        s.bind(our_address)
+        s.listen(1)
+        self._add_to_database(s)
+        self.logger.info("Created server on address '%s:%s'" % our_address)
+
+    def _add_to_database(self, s, state=SERVER, peer=None):
+        """
+                Add the socket to the database, structre will be:
+                {socket :  {
+                "buff":buffer will contain the data we need to send,
+                "state": state of the socket, CLOSE, SERVER or CLIENT,
+                "file": file_object to send,
+                "peer": CLIENT, the server we are connected to,
+                        SERVER, list of the sockets that connected to server
+                        CLOSE, None
+                }
+        """
+        if state not in (CLOSE, SERVER, CLIENT):
+            raise RuntimeError("State not found")
+        self._database[s] = {
+            "state": state,
+        }
+        entry = self._database[s]
+        if state == SERVER:
+            entry["peer"] = []
+        elif state == CLIENT:
+            entry["client"] = Client.Client(s, self._buff_size)
+            entry["peer"] = peer
+        else:
+            self._database[s]["peer"] = None
+        self.logger.debug("s added to database, {'%s': '%s'}" % (
+            s,
+            self._database[s]
+        ))
+
+    def _close_socket(self, s):
+        """ close the socket, remove it from it's from the database and remove
+        references to it from it's peer"""
+        self.logger.debug(self._database)
+        s.close()
+        self._database.pop(s)
+        self.logger.debug("Close success")
+
+    def _change_to_close(self, s):
+        """Change the socket s to close state"""
+        if s in self._database.keys():
+            entry = self._database[s]
+            if entry["state"] == CLIENT:
+                self.logger.debug(entry)
+                if (entry["peer"] is not None and
+                        entry["peer"] in self._database.keys()):
+                    self._database[entry["peer"]]["peer"].remove(s)
+                entry["peer"] = None
+                entry["buff"] = entry["client"].get_send_buff()
+                entry.pop("client")
+            elif entry["state"] == SERVER:
+                self.logger.debug(self._database)
+                for peer_s in self._database[s]["peer"]:
+                    self.logger.debug(peer_s)
+                    self._change_to_close(peer_s)
+                entry["buff"] = ""
+            entry["state"] = CLOSE
+
+    def _build_select(self):
```

Comment:
Why? I already have select and if I'll use the server on Windows I'll the server will be much slower on Windows. I learned how to use select and implemted it.